anyway/parsers/location_extraction.py

```diff
@@ -315,3 +319,26 @@ def extract_geo_features(db, newsflash: NewsFlash) -> None:
         )
         for k, v in location_from_db.items():
             setattr(newsflash, k, v)
+
+
+def try_rectify_location_string(location_string):
+    """
+    This will try to get a gmap location in cases where google can't extract a response from the original
+    sentence but will be able to from a sub-sentence
+    e.g.
+    "גבר נהרג בתאונת דרכים בגליל התחתון"
+    ->no results
+    "בתאונת דרכים בגליל התחתון"
+    ->results
+    """
+
+    # find first location hint and trim prefix
+    trimmed_location_string = ""
+    found = False
+    for word in location_string.split():
+        if word.startswith("ב"):
+            found = True
+        if found:
+            trimmed_location_string += " "
+            trimmed_location_string += word
```

Comment:
You have a start, but not an end. I would suggest something like `word.endswith('.')`.
Also, `token` is slightly more standard name for an item of a split string.

Reply:
the terminology "word" is specific here, because the split by whitespace, will, de-factor tokenize by words

Reply:
It is not important, but "word" is a linguistic term, something with a meaning, whereas "token" is usually just a sequence of non-whitespace characters, including puntuation, for example, which is what you get here.

This is an exreme case of nitpicking though :)